git-discover/tests/upwards/ceiling_dirs.rs

```diff
@@ -31,6 +31,47 @@ fn git_dir_candidate_within_ceiling_allows_discovery() -> crate::Result {
     Ok(())
 }
 
+#[test]
+fn git_dir_candidate_with_cwd_ceiling_ignores_ceiling_if_no_ceiling_dir_may_be_matched() -> crate::Result {
+    let work_dir = repo_path()?;
+    let dir = work_dir.join("some/very/deeply/nested/subdir");
+    // the ceiling dir is equal to the input dir, which doesn't contain a repository.
+    // But we can ignore that just like git does (see https://github.com/Byron/gitoxide/pull/723 for more information)
+    let (repo_path, _trust) = git_discover::upwards_opts(
+        dir.clone(),
+        Options {
+            ceiling_dirs: vec![dir],
+            match_ceiling_dir_or_error: false,
```

Comment:
Thanks to this this should work, and due to the special case it ignores matching ceiling dirs if they are equal to one of the ceiling dirs.

Git does it like this, yes, and `gitoxide` should do the same. I just try to make it work within the logic framework that exists and right now *something* seems off.